### DISCUSSION TITLE
Textadept 11.2 beta 2

### DISCUSSION THREAD

**Discussion by orbitalquark** [orbitalquark/textadept#102] [General]
Textadept 11.2 beta 2 is released from https://github.com/orbitalquark/textadept/releases/tag/textadept_11.2_beta_2

This release improves Textadept's interactions with external tools like code formatters when it comes to undo/redo, as well as filtering text through shell commands.

If you prefer to build Textadept from source, please see the note at the end of this announcement.

Bugfixes:

* Fixed some file extension conflicts.
* Restore view scroll state after `textadept.editing.filter_through()`.
* Do not clobber user's terminal on initialization error.
* Modules: Refresh debugger breakpoints when buffer content is replaced and persist breakpoints and watchpoints on reset.
* Scintilla: Respect system font settings like antialiasing.
* Scintilla: Fix primary selection on GTK 3 and Wayland.

Changes:

* Save/restore view state when undoing/redoing full-buffer changes (e.g. code formatting).
* Added ability to specify find & replace pane font via `ui.find.entry_font`.
* Replaced `events.FILE_BEFORE_RELOAD` and `events.FILE_AFTER_RELOAD` with `events.BUFFER_BEFORE_REPLACE_TEXT` and `events.BUFFER_AFTER_REPLACE_TEXT`.
* Added support for Gleam.
* Scintilla: Added `view.indic_stroke_width`, `view.marker_fore_translucent`, `view.marker_back_translucent`, `view.marker_back_selected_translucent`, and `view.marker_stroke_width`.
* Scintilla: Added new EOL annotation styles.
* Scintilla: Added `view.element_color` for setting UI element colors (e.g. selection, caret, etc.) and deprecated `view:set_sel_fore()`, `view.sel_alpha`, `view.caret_fore`, etc. Also added `view.element_allows_translucent`, `view.element_base_color`, and `view.element_is_set`.
* Scintilla: `view.MARK_CHARACTER` markers now support unicode characters.
* Scintilla: added `view.selection_layer`, `view.caret_line_layer`, and `view.marker_layer` in conjunction with `view.element_color` for alpha transparency.
* Scintilla: Included modifiers in `events.INDICATOR_RELEASE`.
* Scintilla: Update to Unicode 13.
* Updated to Scintilla 5.0.3.

Note: This release uses a pre-release version of Scintilla 5.0.3 which is scheduled for release in a couple of days (~3 Jun). Until then, if you want to build from source, you can use the scintilla503.tgz from https://github.com/orbitalquark/textadept-build or clone Scintilla's repository into *src/scintilla*. Once Scintilla 5.0.3 is released, the usual `make deps` will download it and this notice becomes irrelevant.

**Comment by orbitalquark** [comment]
Scintilla 5.0.3 has been released, so those compiling from source need not do anything different.

**Comment by gabdub** [comment]
I can't download beta2 from  https://github.com/orbitalquark/textadept/releases/tag/textadept_11.2_beta_2.
I get a "Not Found" (Windows) or 404 error page (Linux).

**Comment by orbitalquark** [comment]
Sorry about that. The link has been fixed.

**Comment by orbitalquark** [comment]
For those using the default themes, I just pushed a fix[1] for some selection display issues when the current view is unfocused.

As a side-effect, syntax highlighting is preserved through selections in those themes. Let me know if you encounter any issues.

[1]: https://github.com/orbitalquark/textadept/commit/cb080a63dc6855409300b02195a9d154845ca66d
